### PR TITLE
If specified use folder for log files

### LINKED ISSUE #8: MR #7 breaks REMIND inputdata generation
https://github.com/remindmodel/pre-processing/blob/742a6009bc8ea66b1d9590cf46e50466fdd876b4/start.R#L41
makes
```
Error:
! Settings are missing in provided cfg (cfg$cachefolder)!
Backtrace:
    ▆
 1. └─gms::check_config(cfg, reference_file = "config/APT.cfg", modulepath = NULL)
Execution halted
```

### PR DESCRIPTION
**Pull request by dklein-pik**
so they don't spam into the main folder (useful for APT runs)
Fixes #8 and #10

### CODE REVIEW

File: start.R [hunk 1]
```diff
@@ -38,7 +38,7 @@ if (length(argv) > 0) {
 }
 
 # read in configuration
-cfg <- gms::check_config(cfg, reference_file="config/APT.cfg", modulepath = NULL)
+cfg <- gms::check_config(cfg, reference_file = cfg, modulepath = NULL)
```
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [inline start.R]: Honestly, what was the point of calling `gms::check_config()` in the first place?
> Checks a model configuration file for consistency by comparing it to a reference config file and the given module structure of the model. The function will throw out an error if settings are missing in the config which exist in the reference config, of if settings are set in the config which do not exist in the reference config file or if a realization is chosen for a module which does not exist, not allowed setting combinations.

Comment (dklein-pik) [reply start.R]: My knowledge gap, how I can load the APT.cfg differently, which in turn sources the default.cfg.

Comment (dklein-pik) [reply start.R]: If I remember correctly.

Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply start.R]: Translation:
> The function will throw out an error if settings are missing in `cfg` which exist in `cfg`, of if settings are set in `cfg` which do not exist in `cfg`.

Just cut it.

Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply start.R]: Or if you feel it serves any purpose, set up a `config/reference.cfg` with all the settings deemed legal for input data generation
```
cfg$cachefolder <- NA
cfg$cachetype <- NA
cfg$dev <- NA
cfg$mappinglist <- NA
cfg$renv <- NA
cfg$revision <- NA
```

Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply start.R]: But then you need all those settings in both `config/default.cfg` and `config/APT.cfg`.  For no other reason as to make a function designed for a completely different purpose work.

Comment (dklein-pik) [reply start.R]: > Translation:
> 
> > The function will throw out an error if settings are missing in `cfg` which exist in `cfg`, of if settings are set in `cfg` which do not exist in `cfg`.
> 
> Just cut it.

What does 'cut' mean?

Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply start.R]: Delete.

Comment (dklein-pik) [reply start.R]: You mean I should simply `source(cfgFile)`? Maybe its better to still use `check_config`, it at least checks, if the file sourced has the structure of a config file as expected.

File: start.R [hunk 2]
```diff
@@ -67,18 +67,20 @@ stoppedWithError <- tryCatch({
   return(TRUE)
 })
 
+today <- format(Sys.time(), "%Y-%m-%d")
```
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [inline start.R:70]: OK, does not throw an error anymore.  But now it reports "APT done" for every run, not just APTs.

Comment (dklein-pik) [reply start.R:70]: I am really not in good shape anymore.

File: start.R (hunk 1 above)
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply start.R]: No.  It checks whether `cfgFile` defines the same (no more, no less) settings as … `cfgFile`.  The function will never throw an error, no matter what you put into the .cfg file.
Its completely pointless.

Comment (dklein-pik) [reply start.R]: Ok, I replaced it with `source(cfg)`, agree?

**Pull request merged**